pkg/gke/util/validators.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { get } from '@shell/utils/object';
+import ipaddr from 'ipaddr.js';
+
+// no need to try to validate any fields if the user is still selecting a credential and the rest of the form isn't visible
+export const needsValidation = (ctx: any): Boolean => {
+  return !!ctx.isAuthenticated;
+};
+
+export const requiredTranslation = (ctx: any, labelKey = 'Value'): String => {
+  return ctx.t('validation.required', { key: ctx.t(labelKey) });
+};
+
+export const requiredInCluster = (ctx: any, labelKey: string, clusterPath: string) => {
+  return () :String | undefined => {
+    return needsValidation(ctx) && clusterPath && !get(ctx.normanCluster, clusterPath) ? requiredTranslation(ctx, labelKey) : undefined;
+  };
+};
+
+export const clusterNameChars = (ctx: any ) => {
+  return () :string | undefined => {
+    const { name = '' } = get(ctx, 'normanCluster');
+    const nameIsValid = name.match(/^[a-z0-9\-]*$/);
+
+    return !needsValidation(ctx) || nameIsValid ? undefined : ctx.t('gke.errors.clusterNameChars');
+  };
+};
+
+export const clusterNameStartEnd = (ctx: any) => {
+  return () :string | undefined => {
+    const { name = '' } = get(ctx, 'normanCluster');
+    const nameIsValid = (!!name.match(/^([a-z0-9])+.*([a-z0-9])+$/) || !name.length);
+
+    return !needsValidation(ctx) || nameIsValid ? undefined : ctx.t('gke.errors.clusterNameStartEnd');
+  };
+};
+
+export const ipv4WithCidr = (ctx: any, labelKey: string, clusterPath: string) => {
+  return (val: string) :string | undefined => {
+    let toValidate = get(ctx.normanCluster, clusterPath);
+
+    if (!toValidate) {
+      toValidate = val || '';
+    }
+    let isValidCIDR = false;
+    let isIpv4 = false;
+
+    try {
+      isValidCIDR = ipaddr.isValidCIDR(toValidate);
```

Comment:
It looks like our `isValidCIDR` applies only to ipv4, whereas in this case, we need to check both ipv4 and ipv6 + cidr. I looked into ipv6 regex to do so and they are...quite something. Daunting in length. Bringing in a small library with no deps and a maintainer who appears to be at least paying attention to it if not _frequently_ updating seemed better than copy-pasting a regex I don't comprehend.  Tradeoffs either way, I could be persuaded otherwise.